### PR TITLE
Added data styling with a math expression

### PR DESCRIPTION
**Pull request by mapsdroid**
https://github.com/mapbox/mapbox-android-demo/issues/616

Added a demo that showcases the use of Android's `Expression` to style data.

The demo:
1. Uses weather data of locations, from a GeoJSON file which contains temperature data in Celcius.
2. The data is added to the map using a `GeoJsonSource` 
2. `GeoJsonSource` is rendered using a `SymbolLayer`
3. The celcius data is converted to kelvin using a math `Expression` which is set on the `SymbolLayer`.
4. Tested by running the application on Moto E, Nougat 7.1.1

TODO:
1. Use the correct PNG
2. Remove changes in build.gradle
3. Place text at proper location
4. Add temperature unit to the marker text

### CODE REVIEW

**Comment by mapsdroid** [comment]
FYI @zugaldia @tobrun

File: MapboxAndroidDemo/build.gradle [hunk 1]
```diff
@@ -85,9 +92,17 @@ dependencies {
...
+    //api (dependenciesList.mapboxMapSdk)
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    //implementation 'com.android.support:appcompat-v7:25.4.0'
+    implementation 'com.mapbox.mapboxsdk:mapbox-android-sdk:6.0.0-beta.1@aar'
+    implementation "com.mapbox.mapboxsdk:mapbox-android-telemetry:2.2.9@aar"
+    implementation "com.mapbox.mapboxsdk:mapbox-java-geojson:2.2.9@jar"
+    implementation "com.mapbox.mapboxsdk:mapbox-java-core:2.2.9@jar"
+    implementation "com.squareup.okhttp3:okhttp:3.9.1"
+    implementation "com.google.code.gson:gson:2.8.0"
+    implementation "com.jakewharton.timber:timber:4.5.1"
```
Comment (tobrun) [inline MapboxAndroidDemo/build.gradle]: before releasing need to update to beta.2

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MathExpressionActivity.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+    mapboxMap.addImage(
+        "my-marker-image",
+        BitmapFactory.decodeResource(MathExpressionActivity.this.getResources(),
+            R.drawable.mapbox_marker_icon_default)
+    );
+
+    SymbolLayer markers = new SymbolLayer("marker-layer", source.getId())
+      .withProperties(
+          textField(Expression.toString(celciusToKelvin(Expression.get("temperature")))),
```
Comment (tobrun) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MathExpressionActivity.java]: using static imports for the expressions would improve readability of the DSL

File: MapboxAndroidDemo/build.gradle (hunk 1 above)
Comment (langsmith) [reply MapboxAndroidDemo/build.gradle]: Bumped to current `4.7.0` version

**Pull request closed without merging**